hyperspy/_signals/complex_signal.py

```diff
@@ -0,0 +1,252 @@
+# -*- coding: utf-8 -*-
+# Copyright 2007-2016 The HyperSpy developers
+#
+# This file is part of  HyperSpy.
+#
+#  HyperSpy is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+#  HyperSpy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with  HyperSpy.  If not, see <http://www.gnu.org/licenses/>.
+
+
+import numpy as np
+import h5py
+
+from hyperspy.signal import BaseSignal
+from hyperspy._signals.signal1d import Signal1D
+from hyperspy._signals.signal2d import Signal2D
+
+
+class ComplexSignal(BaseSignal):
+    """BaseSignal subclass for complex data."""
+
+    _dtype = "complex"
+
+    @property
+    def real(self):
+        """Get/set the real part of the data. Returns an appropriate HyperSpy signal."""
+        real = self._deepcopy_with_new_data(np.real(self.data))
+        if real.metadata.General.title:
+            title = real.metadata.General.title
+        else:
+            title = 'Untitled Signal'
+        real.metadata.General.title = 'real({})'.format(title)
+        real.set_signal_type('')  # Result is no longer complex!

```

Comment:
Why setting the signal type? If it is the real part of a DF, it is still a DF. Wouldn't it be better to simply call `assign_subclass`? If the `data` is now real, it should assign the correct (real) subclass...